### PR TITLE
Ovirt attach host devices

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

Add attaching ovirts VMs to host devices

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

ovirt
##### ANSIBLE VERSION

```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/mnecas/Desktop/Projects/Redhat/ansible/lib/ansible/modules/cloud/ovirt']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.5 (default, Apr  4 2018, 15:01:18) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/cloud/ovirt/ovirt_vms.py [hunk 1]
```diff
@@ -860,6 +871,18 @@
       protocol:
         - spice
         - vnc
+
+# Default value of state is present
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vms.py]: Default value of host_device state is present

File: lib/ansible/modules/cloud/ovirt/ovirt_vms.py [hunk 2]
```diff
@@ -860,6 +871,18 @@
...
+- name: Attach host device to VM
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vms.py]: This should be formmated as follows:

```yaml
- name: Attach host devices to virtual machine
  ovirt_vm:
    name: myvm
    host: myhost
    placement_policy: pinned
    host_devices:
      - name: pci_0000_00_06_0
      - name: pci_0000_00_07_0
        state: absent
      - name: pci_0000_00_08_0
        state: present
```

**Review by mwperina — APPROVED** [review]
shipit

File: lib/ansible/modules/cloud/ovirt/ovirt_vms.py [hunk 3]
```diff
@@ -270,6 +270,17 @@
             - Name of the storage domain this virtual machine lease reside on.
             - NOTE - Supported since oVirt 4.1.
         version_added: "2.4"
+    host_devices:
+        description:
+            - Single Root I/O Virtualization - technology that allows single device to expose multiple endpoints that can be passed to VMs
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vms.py]: In this documentation we need to describe the feature a bit not so much detailed.
The first line is enough, the most important part is to define the structure of this variable, that it's list and it takes dictinary as a items and inside dictinary you can define `name` which is name of the device and the `state` where user can specify if it should be attached or detach of the VM.

File: lib/ansible/modules/cloud/ovirt/ovirt_vms.py [hunk 4]
```diff
@@ -273,10 +273,14 @@
...
             - "Enables a virtual machine to be customized to its own compatibility version. If
-            `C(custom_compatibility_version)` is set, it overrides the cluster's compatibility version
+            'C(custom_compatibility_version)' is set, it overrides the cluster's compatibility version
             for this particular virtual machine."
         version_added: "2.7"
-
+    host_devices:
+        description:
+            - Single Root I/O Virtualization - technology that allows single device to expose multiple endpoints that can be passed to VMs
+            - host_devices is an list which contain dictinary with name and state of device
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_vms.py:254]: Take a look how for example `cpu_pinning` is documented, and please document it the same way.

**Comment by machacekondra** [comment]
Fixes: https://github.com/ansible/ansible/issues/34803

**Pull request closed without merging**